packages/docusaurus-theme-classic/src/theme/DocSidebarItem/Category/index.tsx

```diff
@@ -144,6 +160,10 @@ export default function DocSidebarItemCategory({
     }
   }, [collapsible, expandedItem, index, setCollapsed, autoCollapseCategories]);
 
+  if (onlyInactiveUnlistedLinks(activePath, items)) {
+    return null;
+  }
+
```

Comment:
Move call to `isInactiveUnlistedSidebarItem()` upper in the tree? It's better to not render `DocSidebarItemCategory` at all rather than to make it render null inside. We don't necessarily want the hooks above to run in this case